addons/website_sale/views/templates.xml

```diff
@@ -65,14 +65,14 @@
     </template>
 
     <template id="search" name="Search Box">
-        <form t-att-action="keep('/shop'+ ('/category/'+slug(category)) if category else None, search=0)" method="get" t-att-class="_classes">
```

Comment:
@qsm-odoo Question: is it allowed to do such a thing in my case ? Everything is working as I tested (+ runbot green) but I am wondering if it is a good thing to do that. Plus, I did what you told me and everything should be ok. I need your opinion on that so I can end this and you can finally review all of this.

Reply:
It would not make sense to use 'keep' and 'category' in your snippet. So your snippet should only have `action="/shop"`. But here, on the shop page, you can transfer a custom `action`. Inside the t-call, put 
`<t t-set="action" t-value="keep(...)"/>` and in your form template use the `action` variable to set the action (and default to '/shop' if there is no action variable).